content/en/docs/usage/make_build.md

```diff
@@ -0,0 +1,116 @@
+---
+title: Make-based build
+date: 2020-01-11T14:09:21+09:00
+draft: false
+weight: 409
+---
+
+## Building Unikraft via `Makefile`
+
+To manually set up the build environment for the [helloworld app](https://github.com/unikraft/app-helloworld), we first have to create a specific directory structure:
+
+```text
+my-unikernel/
+	|-- apps
+	|   `-- app-helloworld          <- helloworld application
+	|-- libs                        <- additional libraries go here
+	`-- unikraft                    <- Unikraft core
+```
+
+We can do this by executing the following sequence of commands:
+
+```console
+$ mkdir my-unikernel
+$ cd my-unikernel
+$ git clone https://github.com/unikraft/unikraft
+$ mkdir libs
+$ mkdir apps
+$ cd apps
+$ git clone https://github.com/unikraft/app-helloworld
+$ cd app-helloworld
+```
+
+All following configuration and build operations are done from within the `helloworld` application folder.
+Notice that it already contains a `Makefile` that looks like this:
+
+```Makefile
+UK_ROOT ?= $(PWD)/../../unikraft
+UK_LIBS ?= $(PWD)/../../libs
+LIBS :=
+
+all:
+	@$(MAKE) -C $(UK_ROOT) A=$(PWD) L=$(LIBS)
+
+$(MAKECMDGOALS):
+	@$(MAKE) -C $(UK_ROOT) A=$(PWD) L=$(LIBS) $(MAKECMDGOALS)
+```
+
+In the first two lines, we provide the paths to the Unikraft core repository and the directory where we can put additional libraries.
+Since the `helloworld` application uses [`nolibc`](https://github.com/unikraft/unikraft/tree/staging/lib/nolibc) as C library, which comes with the Unikraft core, we do not have any external dependencies.
+We thus do not have to download or reference any libraries in the `Makefile` and can leave the `LIBS` variable empty.
+
+With the `Makefile` present and the directory structure correctly set up, we are able to configure the unikernel as the next step.
+This will prompt a text-based user interface that will allow us to select the target architecture (e.g., `aarch64` or `x86-64`) and platform (e.g., `KVM` or `Xen`).
+
+```console
+$ make menuconfig
+```
+
+By default the build tool selects `x86-64` as target architecture, so we only have to choose a target platform.
+We will decide for `KVM` by navigating to the `Platform Configuration | KVM guest` option and selecting it.
+Afterwards, we save and exit the configuration menu.
+
+You can then build the unikernel with:
+
+```console
+$ make
+```
+
+Consider adding `-jX` with `X` being the number of CPUs in your machine to speed up the build process.
+
+When the build completes you will have a Unikraft-based kernel image (`build/app-helloworld_kvm-x86_64`) that you can start with `QEMU/KVM` using the `-kernel` option.
+See [here](docs/operations/plats/kvm/#build-and-run-unikraft-on-kvm-using-make-and-qemu) for further information on running the unikernel with `KVM`.
+
+## Adding external dependencies
+
```

Comment:
The section above will have ended with a description of `Makefile` and `Makefile.uk`.

Build up on that and say something like: "The build above was the simplest one we can do with Unikraft - configuring, building and running `app-helloworld`. Often, we want to extend the build:
- Add external libraries. No external libraries were part of the build.
- Add new source code files. Only `main.c` was used.
- Use different configuration options.

We detail them below."

The mention how you update `Makefile` to add new libraries, how you update `Makefile.uk` to add new source code files and how you use `Config.uk` to add configuration options (not all) when doing `make menuconfig`.

Reply:
This should be short and point for detailed information to the [`Porting` page](https://github.com/unikraft/docs/blob/main/content/en/docs/develop/porting.md), that will be updated by PR #126